File: task-03/sub-task03/Haskell/pattern.hs
printPattern :: Int -> IO ()
printPattern rows = do
  forM_ [0..rows-1] $ \i -> do
    forM_ [0..rows-i-1] $ \_ -> putStr " "
    forM_ [0..2*i] $ \_ -> putStr "*"
    putStrLn ""
  forM_ [rows-2,rows-3..0] $ \i -> do
    forM_ [0..rows-i-1] $ \_ -> putStr " "
    forM_ [0..2*i] $ \_ -> putStr "*"
    putStrLn ""

main :: IO ()
main = printPattern 3

File: task-03/sub-task03/Python/pattern.py
def print_pattern(rows):
  for i in range(rows):
    for j in range(rows - i):
      print(" ", end="")
    for k in range(2 * i + 1):
      print("*", end="")
    print()

  for i in range(rows - 2, -1, -1):
    for j in range(rows - i):
      print(" ", end="")
    for k in range(2 * i + 1):
      print("*", end="")
    print()

rows = 3
print_pattern(rows)